futures/tests/macro-tests/src/main.rs

```diff
@@ -65,4 +65,37 @@ fn main() {
             _ = b => unreachable!(),
         };
     });
+
+    // stream_select! macro
+    let _ = block_on(async {
+        let endless_ints = |i| stream::iter(vec![i].into_iter().cycle());
+
+        let mut endless_ones = futures04::stream_select!(endless_ints(1i32), stream::pending());
```

Comment:
This module is only used to check things related to macro re-export. So, could you move the test to `futures/tests/async_await_macros.rs`?